### DISCUSSION TITLE
Multi-level joining in HTTP API

### DISCUSSION THREAD

**Discussion by EnzoBitMatrix** [supabase/supabase#5484] [Questions] [answered]
Hoping somebody can help me with this, I will try and explain what I have.

We need to be able to get records from three layers deep via the HTTP REST api.

We can get two layers deep without a problem.

Our current REST API is:
`https://SUB.supabase.co/rest/v1/test_user_mission?user_id=eq.{user_id}&select=*,test_station(*),test_mission(*)`

This returns a JSON of:
```
[
    {
        "id": 1,
        "user_id": 1,
        "user_station_id": 1,
        "mission_id": 1,
        "test_station": {
            "id": 1,
            "station_name": "Test Station",
            "station_desc": "This is the test station"
        },
        "test_mission": {
            "id": 1,
            "mission_date": "2022-02-16T01:37:25+00:00",
            "scenario_id": 1,
            "director_id": 1
        }
    }
]
```
which works exactly as we and we like it.

However, there are two more tables at play... test_scenario and test_director

In the 'test_mission' section above, there is the scenario_id and director_id.  How do we look these up in the URL?

So, we are hoping that the end JSON will look like this:
```
[
    {
        "id": 1,
        "user_id": 1,
        "user_station_id": 1,
        "mission_id": 1,
        "test_station": {
            "id": 1,
            "station_name": "Test Station",
            "station_desc": "This is the test station"
        },
        "test_mission": {
            "id": 1,
            "mission_date": "2022-02-16T01:37:25+00:00",
            "test_scenario": {
                     "id": 1,
                      "scenario_name": "Test Scenario",
                      "scenario_desc": "This is just a test"
             },
            "test_director":{ 
                      "director_id": 1,
                      "director_name": "Joe Bloggs",
                      "director_email": "joe@bloggs.com"
             },
        }
    }
]
```

I may have stuffed up the actual formatting above, but you get the idea, we want to be able to select * from the two additional by foreignkeys in the test_mission table.

I really hope I have explained that well enough.  Any assistance on how we can achieve this would be greatly appreciated!  We are new to Supabase and REST API's

**Comment by steve-chavez** [comment, marked as answer]
Try it like:

```http
GET /test_user_mission?user_id=eq.{user_id}&select=*,test_station(*),test_mission(*,test_scenario(*),test_director(*))
```

**Reply by EnzoBitMatrix** [reply]
Wow, thank you so much!
That worked perfectly - I've spent ages trying to get it to work.

You're a legend!
